src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -415,13 +468,118 @@ public void setInSelectCartesianProductThreshold(int warn, int fail)
+    }
+
+    @Override
+    public String getReadConsistencyLevelsWarnedCSV()
+    {
+        return toCSV(DEFAULT_CONFIG.getReadConsistencyLevelsWarned(), ConsistencyLevel::toString);
+    }
+
+    @Override
+    public void setReadConsistencyLevelsWarned(Set<ConsistencyLevel> consistencyLevels)
+    {
+        DEFAULT_CONFIG.setReadConsistencyLevelsWarned(consistencyLevels);
+    }
+
+    @Override
+    public void setReadConsistencyLevelsWarnedCSV(String consistencyLevels)
+    {
+        DEFAULT_CONFIG.setReadConsistencyLevelsWarned(fromCSV(consistencyLevels, ConsistencyLevel::fromString));
+    }
+
+    @Override
+    public Set<ConsistencyLevel> getReadConsistencyLevelsDisallowed()
+    {
+        return DEFAULT_CONFIG.getReadConsistencyLevelsDisallowed();
+    }
+
+    @Override
+    public String getReadConsistencyLevelsDisallowedCSV()
+    {
+        return toCSV(DEFAULT_CONFIG.getReadConsistencyLevelsDisallowed(), ConsistencyLevel::toString);
+    }
+
+    @Override
+    public void setReadConsistencyLevelsDisallowed(Set<ConsistencyLevel> consistencyLevels)
+    {
+        DEFAULT_CONFIG.setReadConsistencyLevelsDisallowed(consistencyLevels);
+    }
+
+    @Override
+    public void setReadConsistencyLevelsDisallowedCSV(String consistencyLevels)
+    {
+        DEFAULT_CONFIG.setReadConsistencyLevelsDisallowed(fromCSV(consistencyLevels, ConsistencyLevel::fromString));
+    }
+
+    @Override
+    public Set<ConsistencyLevel> getWriteConsistencyLevelsWarned()
+    {
+        return DEFAULT_CONFIG.getWriteConsistencyLevelsWarned();
+    }
+
+    @Override
+    public String getWriteConsistencyLevelsWarnedCSV()
+    {
+        return toCSV(DEFAULT_CONFIG.getWriteConsistencyLevelsWarned(), ConsistencyLevel::toString);
+    }
+
+    @Override
+    public void setWriteConsistencyLevelsWarned(Set<ConsistencyLevel> consistencyLevels)
+    {
+        DEFAULT_CONFIG.setWriteConsistencyLevelsWarned(consistencyLevels);
+    }
+
+    @Override
+    public void setWriteConsistencyLevelsWarnedCSV(String consistencyLevels)
+    {
+        DEFAULT_CONFIG.setWriteConsistencyLevelsWarned(fromCSV(consistencyLevels, ConsistencyLevel::fromString));
+    }
+
+    @Override
+    public Set<ConsistencyLevel> getWriteConsistencyLevelsDisallowed()
+    {
+        return DEFAULT_CONFIG.getWriteConsistencyLevelsDisallowed();
+    }
+
+    @Override
+    public String getWriteConsistencyLevelsDisallowedCSV()
+    {
+        return toCSV(DEFAULT_CONFIG.getWriteConsistencyLevelsDisallowed(), ConsistencyLevel::toString);
+    }
+
+    @Override
+    public void setWriteConsistencyLevelsDisallowed(Set<ConsistencyLevel> consistencyLevels)
+    {
+        DEFAULT_CONFIG.setWriteConsistencyLevelsDisallowed(consistencyLevels);
+    }
+
+    @Override
+    public void setWriteConsistencyLevelsDisallowedCSV(String consistencyLevels)
+    {
+        DEFAULT_CONFIG.setWriteConsistencyLevelsDisallowed(fromCSV(consistencyLevels, ConsistencyLevel::fromString));
+    }
+
```

Comment:
@adelapena I'm trying to make sense of the coexistence of `DEFAULT_CONFIG` and `CONFIG_PROVIDER`. The former seems like it's just `CONFIG_PROVIDER.getOrCreate(null)` for all intents and purposes...unless we actually use `custom_guardrails_config_provider_class`, at which point `DEFAULT_CONFIG` becomes incorrect, right?

Reply:
`CONFIG_PROVIDER` is used to get the actual config that is going to be used by the guardrails. This config can depend on the user if there is a custom implementation. If such custom implementation doesn't exist, then the default config would be the one defined in `cassandra.yaml`, which is accessed through either `DEFAULT_CONFIG` or `DatabaseDescriptor.getGuardrailsConfig()`.

However, `CONFIG_PROVIDER.getOrCreate(null)` doesn't necessarily return `DEFAULT_CONFIG`, since a custom implementation of the config provider can return something different, while `DEFAULT_CONFIG` is always the configuration defined in `cassandra.yaml`.

As for the JMX setters/getters, they are meant to exclusively change the default configuration defined in `cassandra.yaml`, which is accessible through either `DEFAULT_CONFIG` or `DatabaseDescriptor.getGuardrailsConfig()`. Indeed, the `GuardrailsConfig` interface doesn't even have setters. Any custom implementation interested on similar JMX getters/setters should provide its own mbeans, since the ones defined on `GuardrailsMBean` are only for the default config.

This is done this way to keep the JMX methods in `GuardrailsMBean` independent of guardrail config pluggability in general, and of `ClientState` in particular. This is so partly because that pluggability mechanism is not yet officially supported. This was agreed on CASSANDRA-17147.

Reply:
Thanks for the context.